lib/promiscuous/subscriber/worker/distributor.rb

```diff
@@ -0,0 +1,88 @@
+class Promiscuous::Subscriber::Worker::Distributor
+  def initialize(root)
+    @root = root
+  end
+
+  def start
+    num_threads = Promiscuous::Config.subscriber_threads
+    Promiscuous::Config.subscriber_topics.each do |topic|
+      @distributor_threads ||= num_threads.times.map { DistributorThread.new(topic) }
```

Comment:
In case you call start twice by mistake, i guess that's defensive coding but i took the idea from the runner